File: app/src/main/java/social/entourage/android/invite/contacts/InviteItem.java
package social.entourage.android.invite.contacts;

/**
 * Created by mihaiionescu on 28/07/16.
 */
public class InviteItem {

    protected static final int TYPE_CONTACT_NAME = 0;
    protected static final int TYPE_CONTACT_PHONE = 1;
    protected static final int TYPE_SECTION = 2;

    public static final int TYPE_COUNT = 3;

    private int itemType = TYPE_CONTACT_NAME;
    private String itemText;
    private boolean isSelected = false;

    public InviteItem(int itemType, String itemText) {
        this.itemType = itemType;
        this.itemText = itemText;
    }

    public int getItemType() {
        return itemType;
    }

    public String getItemText() {
        return itemText;
    }

    public boolean isSelected() {
        return isSelected;
    }

    public void setSelected(final boolean selected) {
        isSelected = selected;
    }

}

File: app/src/main/java/social/entourage/android/entourage/invite/contacts/InviteItem.kt
package social.entourage.android.entourage.invite.contacts

/**
 * Created by mihaiionescu on 28/07/16.
 */
open class InviteItem(val itemType: Int, val itemText: String) {
    var isSelected = false

    companion object {
        const val TYPE_CONTACT_NAME = 0
        const val TYPE_CONTACT_PHONE = 1
        const val TYPE_SECTION = 2
        const val TYPE_COUNT = 3
    }

}

class InviteItemContactName(itemText: String, val cursorPosition: Int)
    : InviteItem(TYPE_CONTACT_NAME, itemText)

class InviteItemContactPhone(itemText: String, val cursorPosition: Int)
    : InviteItem(TYPE_CONTACT_PHONE, itemText)

class InviteItemSection(itemText: String)
    : InviteItem(TYPE_SECTION, itemText)